src/io/ExportCSV.cpp

```diff
@@ -35,35 +35,41 @@ struct StridedAccess {
 };
 } // namespace
 
-void ExportCSV::doExport(
-    const std::string &name,
-    const std::string &location,
-    const mesh::Mesh & mesh)
-{
-  PRECICE_TRACE(name, location, mesh.getName());
-  PRECICE_ASSERT(!name.empty());
+ExportCSV::ExportCSV(
+    std::string_view  participantName,
+    std::string_view  location,
+    const mesh::Mesh &mesh,
+    ExportKind        kind,
+    int               frequency,
+    int               rank,
+    int               size)
+    : Export(participantName, location, mesh, kind, frequency, rank, size){};
 
-  // Ignore empty meshes
-  if (mesh.empty()) {
-    return;
-  }
+void ExportCSV::doExport(int index, double time)
+{
+  PRECICE_TRACE(index, time, _mesh->getName());
+  PRECICE_ASSERT(index >= 0);
+  PRECICE_ASSERT(time >= 0.0);
 
-  // Construct full filename
-  std::string filename{name};
-  int         rank{0};
-  if (utils::IntraComm::isParallel()) {
-    rank = utils::IntraComm::getRank();
-    filename.append("_").append(std::to_string(rank));
+  // Construct filename
+  std::string filename;
+  if (isParallel()) {
+    // Participant-Mesh-r2.it2
+    filename = fmt::format("{}-{}-r{}.{}{}.csv", _participantName, _mesh->getName(), _rank, kindPrefix(), index);
+  } else {
+    // Participant-Mesh.it2
+    filename = fmt::format("{}-{}.{}{}.csv", _participantName, _mesh->getName(), kindPrefix(), index);
   }
-  filename.append(".csv");
 
   namespace fs = std::filesystem;
-  fs::path outfile(location);
-  if (not location.empty()) {
+  fs::path outfile(_location);
+  if (not _location.empty()) {
     fs::create_directories(outfile);
   }
   outfile /= filename;
 
+  const auto &mesh = *_mesh;
```

Comment:
Why is this necessary? If you work with a pointer before, I would prefer a pointer in the following as well.

Reply:
It prevents having to change the uses of mesh in the remaining code. I'll clean it up.